src/extension.ts

```diff
@@ -1732,8 +1744,40 @@ async function deleteContextKubernetes(explorerNode: explorer.KubernetesObject)
 }
 
 async function useNamespaceKubernetes(explorerNode: explorer.KubernetesObject) {
-    if (await kubectlUtils.switchNamespace(kubectl, explorerNode.id)) {
-        refreshExplorer();
+    if (explorerNode) {
+        if (await kubectlUtils.switchNamespace(kubectl, explorerNode.id)) {
+            refreshExplorer();
+        }
+    } else {
+        promptKindName([kuberesources.allKinds.namespace], undefined,
```

Comment:
We should pre-fill the namespaces a user may switch to here. Entering the namespace name is clunky.

Reply:
am working on to make the type-search-list (filter) pattern . WDYT?

Reply:
I was having that first but then some times it takes few seconds to load the namespaces list, hence thought to have typing allowed